src/lib/sidenav/sidenav.ts

```diff
@@ -197,6 +215,15 @@ export class MdSidenav implements AfterContentInit {
   }
 
   /**
+   * Handles the keyboard events.
+   */
+  handleKeydown(event: KeyboardEvent) {
+    if (event.keyCode === ESCAPE) {
+      this.close();
```

Comment:
probably want to stop propagation (if we have a dialog with a sidenav, esc should close the sidenav but not the dialog)